File: JavaScript/프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var cur, idx;
    var answer = 0;
    var total = [];
    var queue = [];
    for (let i=0; i<n; i++) total.push(1);
    
    while (total.includes(1)) {
        answer++;
        idx = total.findIndex((e) => e == 1);
        total[idx] = -1
        queue.push(idx);
        while (queue.length > 0) {
            cur = queue.shift();
            for (let i=0; i<n; i++) {
                if (computers[cur][i] == 1 && total[i] == 1) {
                    queue.push(i);
                    total[cur] = -1;
                    computers[cur][i] = 0;
                    computers[i][cur] = 0;
                }
            }
        }
    }
    
    return answer;
}

File: Java/프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        int cur, idx;
        List<Integer> total = new ArrayList<>();
        Deque<Integer> queue = new ArrayDeque<>();
        
        for (int i=0; i<n; i++) total.add(i);
        
        while (!total.isEmpty()) {
            answer++;
            queue.add(total.remove(0));
            while (!queue.isEmpty()) {
                cur = queue.pop();
                for (int i=0; i<n; i++) {
                    if (computers[cur][i] == 1 && total.contains(i)) {
                     queue.add(i);
                     total.remove((Object) i);
                     computers[cur][i] = 0;
                     computers[i][cur] = 0;
                    }
                }
            }
        }
        return answer;
    }
}